data/exploits/vmware_view_planner_4_6_uploadlog_rce/log_upload_wsgi.py

```diff
@@ -0,0 +1,107 @@
+#! /usr/bin/env python3
+import cgi
+import os,sys
+import logging
+import json
+
+WORKLOAD_LOG_ZIP_ARCHIVE_FILE_NAME = "workload_log_{}.zip"
+
+class LogFileJson:
+    """ Defines format to upload log file in harness
+
+    Arguments:
+    itrLogPath : log path provided by harness to store log data
+    logFileType : Type of log file defined in api.agentlogFileType
+    workloadID [OPTIONAL] : workload id, if log file is workload specific
+
+    """
+    def __init__(self, itrLogPath, logFileType, workloadID = None):
+        self.itrLogPath = itrLogPath
+        self.logFileType = logFileType
+        self.workloadID = workloadID
+
+    def to_json(self):
+        return json.dumps(self.__dict__)
+
+    @classmethod
+    def from_json(cls, json_str):
+        json_dict = json.loads(json_str)
+        return cls(**json_dict)
+
+class agentlogFileType():
+    """ Defines various log file types to be uploaded by agent
+
+    """
+    WORKLOAD_ZIP_LOG = "workloadLogsZipFile"
+
+try:
+    # TO DO: Puth path in some config
+    logging.basicConfig(filename="/etc/httpd/html/logs/uploader.log",filemode='a', level=logging.ERROR)
+except:
+    # In case write permission is not available in log folder.
+    pass
+
+logger = logging.getLogger('log_upload_wsgi.py')
+
+def application(environ, start_response):
+    logger.debug("application called")
+
+    if environ['REQUEST_METHOD'] == 'POST':
+        post = cgi.FieldStorage(
+            fp=environ['wsgi.input'],
+            environ=environ,
+            keep_blank_values=True
+        )
+        # TO DO: Puth path in some config or read from config is already available

```

Comment:
Same as above

Reply:
These were all taken from the original Metasploit template code I based this on...looks like I'm going to need to do a review and clean this up. Give me a sec whilst I do that.

Reply:
Actually sorry I was super out of it cause of reviewing another PR and I didn't properly look through this. All the `TODO` comments are there cause this file is a copy of the original file from the vulnerable target, which included the `TODO` comments, but with a section added to allow us to inject our payload and execute it properly.

Reply:
I mean, you could finish the code for them.